### PR TITLE
Avoiding assert in debug mode

### PR DESCRIPTION
**Pull request by loumalouomega**
This is temporal solution, the new gidpost library is giving problems with the new versions of gcc, so instead of updating the whole library I just added that in case of debug mode we avoid the assertions (Jorge from giD group recommended that)

### CODE REVIEW

**Comment by RiccardoRossi** [comment]
well i don't love this fix-

the kratos has a flag
-DDO_NOT_DEFINE_NDEBUG

which explicitly tells not to define NDEBUG. We should not break this beahaviour.

Did u detect what is the exact source of the error?

**Comment by RiccardoRossi** [comment]
Ok, afternoon speaking with the gid team and verificar the branch, i approve the merge

**Comment by loumalouomega** [comment]
@RiccardoRossi You have not approved the merge yet

File: external_libraries/gidpost/CMakeLists.txt [hunk 1]
```diff
@@ -11,6 +11,7 @@ set( GIDPOST_SOURCES
      )
 ##gidpostfor.c gidpostforAPI.c
 
+add_definitions(-DNDEBUG)
```
Comment (pooyan-dadvand) [inline external_libraries/gidpost/CMakeLists.txt:14]: I would add a warning if debug is enabled saying that it ignores the debug flag

**Review by pooyan-dadvand — COMMENTED** [review]
As you mentioned it fixes until gid team gives a solution 
And thanks for changing the error macros!

**Comment by loumalouomega** [comment]
@pooyan-dadvand Message added, I change your "god team" to "gid team" too, I don't know if it was irony, just in case

**Comment by maceligueta** [comment]
By the way, -DDO_NOT_DEFINE_NDEBUG is not the best name ever for a variable. Is it possible to rename it somehow? 
We already have -DKRATOS_DEBUG, can we use it to set or unset NDEBUG? I know that they do different things, but we can unify them anyways, for simplicity.

**Review by RiccardoRossi — APPROVED** [review]

**Pull request merged**